main/zoe/guide/contracts/nft-drop.md

```diff
@@ -0,0 +1,107 @@
+    mint.mintGains(harden({ NFTs: amount }), buyerSeat);
+    currentId += 1n;
+
+    // Take the money
+    sellerSeat.incrementBy(buyerSeat.decrementBy(harden({ Money: pricePerNFT })));
+
+    zcf.reallocate(buyerSeat, sellerSeat);
+    buyerSeat.exit();
+
+    return 'your offer was successful';
+  };
+
+  const publicFacet = Far('NFT Drop', {
+    makeInvitation: () => zcf.makeInvitation(buyNFTs, 'buyNFTs'),
+  });
+
+  return harden({ publicFacet });
+};
+
+```
+
+Taking a look inside of the contract code, you will notice that the minting process occurs **at the time of purchase**. This approach is in contrast to one in which a set of NFTs are minted prior to being made available to the public for purchase. 
+
+To accomplish this design, the contract first creates an internal `mint` variable. Note that the `mint` variable specifies the [`AssetKind`](../../../ertp/api/issuer.md#makeissuerkit-allegedname-assetkind-displayinfo) to be a `SET` which is necessary when creating NFTs with Zoe.
+
+```js
+const mint = await zcf.makeZCFMint(nftName, AssetKind.SET);
+```
+
+Once the `mint` has been created, the next step is to use it within the `buyNFTs` function which will be provided to the the public as an invitation. Additionally, we will see that each NFT has a unique identifier, `currentId`, that is embedded within the NFT at the time it is minted. Although it is intended to be a placeholder, the use of `currentId` is meant to demonstrate how one could add unique properties to an NFT.
+
+```JS
+  /** @type {OfferHandler} */
+  const buyNFTs = (buyerSeat) => {
+    // Mint the NFTs
+    const amount = AmountMath.make(NFTBrand, harden([currentId]));
+    mint.mintGains(harden({ NFTs: amount }), buyerSeat);
+    currentId += 1n;
+
+    // Take the money
+    sellerSeat.incrementBy(buyerSeat.decrementBy(harden({ Money: pricePerNFT })));
+
+    zcf.reallocate(buyerSeat, sellerSeat);
+    buyerSeat.exit();
+
+    return 'your offer was successful';
+  };
+```
+
+The final step is to create a [Remotable](../../../guides/js-programming/far.md#marshaling-by-copy-or-by-presence) for our public invitation using `zcf`s `makeInvitation` method. 
+
+```js
+const publicFacet = Far('NFT Drop', {
+  makeBuyInvitation: () => zcf.makeInvitation(buyNFTs, 'buyNFTs'),
+});
+
+return harden({ publicFacet });
+```
+The contract is ready to be instantiated using the `E` as demonstrated in the test code below. In this snippet, the `nftName` for this collection is `PetRocks`, all of which are sold for the same fixed-price. 
+
+```js
+  const terms = harden({
+    pricePerNFT: AmountMath.make(moneyBrand, 10n),
+    nftName: 'PetRocks',
+  });
+  const { instance } = await E(zoe).startInstance(installation, issuers, terms);
+
+  const publicFacet = E(zoe).getPublicFacet(instance);
+  // we now have access to the pubic invitation for the `buyNFTs` function.
+  const invitation = E(publicFacet).makeInvitation();
```

Comment:
This code would benefit from the testable snippets stuff; it's not in the other repo.

```suggestion
  const invitation = E(publicFacet).makeBuyInvitation();
```

Reply:
Hmmm, are u saying that there's a currently examples of runnable code snippets in the docs? If so, then I'd gladly translate this code into that as i believe it would be greatly beneficial for developers.

Reply:
Nevermind! Seeing the snippets info in the Contributing.md file now 🙂  thanks!

Reply:
yes; see [details in CONTRIBUTING.md](https://github.com/Agoric/documentation/blob/main/CONTRIBUTING.md#importing-and-testing-code-snippets).

Setting up the context to test a contract is kinda involved, so it's perhaps better left to a separate PR.

Reply:
cool! i actually split it up so the commit i just pushed included all updates besides the snippet. 

reading each other's minds !